src/Tools/ExternalAccess/CustomMessage/External/CustomMessageContext.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.CodeAnalysis.CustomMessageHandler;
+
+public sealed class CustomMessageContext
+{
+    internal CustomMessageContext(Document? document, Solution solution)
+    {
+        Document = document;
+        Solution = solution ?? throw new ArgumentNullException(nameof(solution));
```

Comment:
No need to throw. You can also use primary constructor.

Reply:
Ok for the throw.
Not sure how to make an internal primary constructor for a public class though..